test/sqllogictest/outer_join_simplification.slt

```diff
@@ -429,24 +429,8 @@ with mutually recursive
   )
 select * from c0;
 ----
-Explained Query:
-  Return // { keys: "([0])" }
-    Get l0 // { keys: "([0])" }
-  With Mutually Recursive
-    cte l0 =
-      TopK group_by=[#0] limit=1 // { keys: "([0])" }
-        Distinct project=[#0..=#3] // { keys: "([0, 1, 2, 3])" }
-          Union // { keys: "()" }
-            Project (#0..=#2, #0) // { keys: "()" }
-              Join on=(#0 = #3) type=differential // { keys: "()" }
-                ArrangeBy keys=[[#0]] // { keys: "()" }
-                  ReadStorage materialize.public.foo // { keys: "()" }
-                ArrangeBy keys=[[#0]] // { keys: "([0])" }
-                  Project (#0) // { keys: "([0])" }
-                    Get l0 // { keys: "([0])" }
-            Get l0 // { keys: "([0])" }
-
-Source materialize.public.foo
+Explained Query (fast path):
+  Constant <empty>
```

Comment:
Interesting that so many plans in `outer_join_simplification.slt` mostly disappeared. It seems we'll have to get more real about our test queries :)

Reply:
Yeah, it's probably healthy, because we are some amount of relying on naiveté in the optimizer, and while this transform blows away the query, you could imagine others (unique key prop) that more quietly sneaks some information around a loop and gets around something we thought we were testing.

Reply:
I think there are easy fixes that involve bringing a table in at tasteful moments, but I'm also not certain I can recover the intent of the queries perfectly. The comments on them are great, and should get us there, but it will involve some amount of staring at things.

Reply:
> you could imagine others (unique key prop) that more quietly sneaks some information around a loop and gets around something we thought we were testing.

Yes. Similar things have happened when Alex wired in your analysis framework to certain transforms, e.g. https://github.com/MaterializeInc/materialize/pull/26856/commits/1382b5e165c6cd7427e64ffdd47e576a8be53057#diff-cff3ddbf2e3561f6e25def9406f616a6a9039f6ffea0b069e631ae6728ba1a93L485

Also: https://github.com/MaterializeInc/materialize/pull/25591/commits/94c64cd55f03c84d1fe2214c2ca24cd184c57a1c